### PR TITLE
Chat app - Upgraded Version

### LINKED ISSUE #8: Group Chat
When following streamers, be able
to chat with other common followers of the user (similar to the way twitch allows people to chat)

### PR DESCRIPTION
**Pull request by tarelmadar**
I have upgraded the chat app to show messages history and some details about the user, such as username and timestamp.

### CODE REVIEW

**Review by ItzhakiGuy — DISMISSED** [review]
Sending a new comment, I can't find the way to delete a comment :-)

File: sellotape/sellotape_dj/chat/models.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from django.db import models
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+# Create your models here.
+class Message(models.Model):
+    author = models.ForeignKey(User, related_name = 'author_messages', on_delete = models.CASCADE)
```
Comment (ItzhakiGuy) [inline sellotape/sellotape_dj/chat/models.py]: We have extended the regular Django `User` attributes, and we have called it a `Profile`, it should correspond with the `author` attribute on your `models.py` - chat app.
Plus, I think the `Message` object should be on the README.MD as you are creating a new ORM.
See PR #27 , it supposed to clarify the objects modeling.

**Review by ItzhakiGuy — CHANGES_REQUESTED** [review]

File: sellotape/sellotape_dj/chat/templates/chat/index.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<html>
```
Comment (Omeramsc) [inline sellotape/sellotape_dj/chat/templates/chat/index.html]: Is there a specific reason not to make use of the base.html template you guys implemented?

File: sellotape/sellotape_dj/chat/templates/chat/room.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
+{% load static %}
+<html>
```
Comment (Omeramsc) [inline sellotape/sellotape_dj/chat/templates/chat/room.html]: Is there a specific reason not to make use of the base.html template you guys implemented?

File: sellotape/sellotape_dj/chat/views.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from django.contrib.auth.decorators import login_required
+from django.utils.safestring import mark_safe
+from django.shortcuts import render
+import json
+
+def index(request):
```
Comment (Omeramsc) [inline sellotape/sellotape_dj/chat/views.py:6]: You better lint/reformat your code, 
Once you'll implement a CI with a linter, it'll fail in cases like this where there are no 2 lines between functions

File: sellotape/sellotape_dj/chat/views.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    return render(request, 'chat/index.html')
+@login_required
```
Comment (Omeramsc) [inline sellotape/sellotape_dj/chat/views.py:8]: ^same

File: sellotape/sellotape_dj/chat/tests.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (Omeramsc) [inline sellotape/sellotape_dj/chat/tests.py:3]: This is what we call "הכנה למזגן"... if you don't plan to introduce tests on this PR (although you probably should), this file is not needed.

File: sellotape/sellotape_dj/chat/models.py (hunk 1 above)
Comment (Omeramsc) [reply sellotape/sellotape_dj/chat/models.py]: Do note that if you change this, you'd also need a new migration.

File: sellotape/sellotape_dj/chat/views.py (hunk 4 above)
Comment (11galpid) [reply sellotape/sellotape_dj/chat/views.py:6]: We'll add a reformat PR to address this format issues, it happens in other parts of the app aswell.

**Comment by 11galpid** [comment]
Pushed in a few small fixes, doesn't change the general structure of the chat-app

**Comment by 11galpid** [comment]
@tarelmadar Note, the 'Bootstrap the ChatApp' issue will be addressed in a different PR, after the chatapp will be approved formally, its not part of this PR

File: sellotape/sellotape_dj/chat/models.py (hunk 1 above)
Comment (11galpid) [reply sellotape/sellotape_dj/chat/models.py]: @tarelmadar Did you address this?

**Review by ItzhakiGuy — APPROVED** [review]
LGTM :-)

**Review by 11galpid — APPROVED** [review]
Looks good

**Pull request merged**